cluster-autoscaler/core/autoscaler.go

```diff
@@ -50,5 +50,8 @@ func NewAutoscaler(opts AutoscalerOptions, predicateChecker *simulator.Predicate
 		configFetcher := dynamic.NewConfigFetcher(opts.ConfigFetcherOptions, kubeClient, kubeEventRecorder)
 		return NewDynamicAutoscaler(autoscalerBuilder, configFetcher)
 	}
+	if opts.NodeGroupAutoDiscovery != "" {
```

Comment:
If someone passes --configmap and --node-group-auto-discovery the latter will be silently ignored. It would be nice to at least log a warning (and maybe mention it in docs).